### PR TITLE
Allow non-R solution code

### PR DESCRIPTION
**Pull request by gadenbuie**
* Don't try to parse non-R solution code
* `.solution` isn't available in these cases
* Update `mock_this_exercise()` to follow same flow

### CODE REVIEW

File: R/gradethis_exercise_checker.R [hunk 1]
```diff
@@ -169,7 +169,7 @@ check_exercise <- function(
   )
 
   # Skip parse checking if it was already done in {learnr}
-  if (!learnr_includes_parse_check(stage)) {
+  if (!learnr_includes_parse_check(stage) && identical(learnr_args$engine %||% "r", "r")) {
```
Comment (gadenbuie) [inline R/gradethis_exercise_checker.R:172]: In the future, I'll think we'll want to standardize how we discover what language a piece of code is, but this is fine for now

File: R/gradethis_exercise_checker.R [hunk 2]
```diff
@@ -271,6 +271,22 @@ prepare_check_env <- function(learnr_args, envir_caller = rlang::caller_env()) {
   }
 
   # Delayed evaluation of `.solution`
+  engine <- knitr_engine_caption(learnr_args[["engine"]])
+  if (!identical(engine, "R")) {
```
Comment (gadenbuie) [inline R/gradethis_exercise_checker.R:275]: Same here, I'd rather have a separate function or pattern to check for the exercise/code language, but I think that's a larger amount of work we'll want to tackle later on

Comment (rossellhayes) [reply R/gradethis_exercise_checker.R:275]: Quick question: why does `engine == "R"`, but `learnr_args$engine == "r"`?

Comment (gadenbuie) [reply R/gradethis_exercise_checker.R:275]: yeah that's totally one of the reasons to standardize. `knitr_engine_caption()` turns engine strings like `r`, `sql`, `javascript` into text suitable for the exercise caption, e.g. `R`, `SQL` or `JavaScript`. So `"r"` or `NULL` comes out as `"R"`.

Comment (gadenbuie) [reply R/gradethis_exercise_checker.R:275]: (In this case we're prepping the string for use in the message, so it makes sense to compare with `"R"`.)

File: R/mock.R [hunk 3]
```diff
@@ -81,11 +85,11 @@ mock_this_exercise <- function(
   .label = "mock",
   .engine = "r",
   .stage = "check",
+  .result = rlang::missing_arg(),
```
Comment (rossellhayes) [inline R/mock.R:88]: What's the difference between `.result = rlang::missing_arg()` and just `.result` with no default value?

**Review by rossellhayes — APPROVED** [review]
Looks great!

File: R/mock.R (hunk 3 above)
Comment (gadenbuie) [reply R/mock.R:88]: If `.result` isn't provided, we want to create `.result` from the `.user_code`, provided that it's R code that we can evaluate.

If we don't have R code though, we need the user to tell us what the result will be. That result _might be_ `NULL`, so I'm using `rlang::missing_arg()` to tell the difference.

We could leave `.result` without a default value and use `missing(.result)` to catch that, but I prefer being explicit. I also prefer using "no default value" as a strong "required argument" signal; in this case `.result` isn't required.

**Pull request merged**